sympy/sets/sets.py

```diff
@@ -26,9 +26,8 @@
 from sympy.utilities.exceptions import SymPyDeprecationWarning
 from sympy.utilities.iterables import iproduct, sift, roundrobin
 from sympy.utilities.misc import func_name, filldedent
-
 from mpmath import mpi, mpf
-
+from sympy.vector.basisdependent import factor as fctr
```

Comment:
```
from sympy.polys import factor
```
No need to rename as `fctr`

Reply:
I tried using `from sympy.polys import factor` but it shows this error "cannot import name 'factor' from partially initialized module 'sympy.polys".

The previous one was working but it start giving a new issue in different file "ImportError: cannot import name 'sqrt' from partially initialized module 'sympy' ".

Reply:
Maybe you'll need to use a delayed import then and import it in the function. You shouldn't be importing it from vector as it isn't defined there.

Try this:
https://github.com/sympy/sympy/blob/f119ab5d30e8b373dc8956d41b7c11e53c140209/sympy/core/expr.py#L3583-L3586

Reply:
thank you so much this works :)